lib/auth/grpcserver.go

```diff
@@ -4901,6 +4901,102 @@ func (g *GRPCServer) GetHeadlessAuthentication(ctx context.Context, req *proto.G
 	return authReq, trace.Wrap(err)
 }
 
+// CreatePlugin creates a plugin resource.
+func (g *GRPCServer) CreatePlugin(ctx context.Context, plugin *types.PluginV1) (*emptypb.Empty, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return nil, trace.Wrap(auth.CreatePlugin(ctx, plugin))
+}
+
+// DeletePlugin deletes a plugin resource.
+func (g *GRPCServer) DeletePlugin(ctx context.Context, req *types.ResourceRequest) (*emptypb.Empty, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return nil, trace.Wrap(auth.DeletePlugin(ctx, req.Name))
+}
+
+// DeleteAllPlugins deletes all plugin resources.
+func (g *GRPCServer) DeleteAllPlugins(ctx context.Context, _ *emptypb.Empty) (*emptypb.Empty, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return nil, trace.Wrap(auth.DeleteAllPlugins(ctx))
+}
+
+// ListPlugins lists plugin resources.
+func (g *GRPCServer) ListPlugins(ctx context.Context, req *types.ListPluginsRequest) (*types.ListPluginsResponse, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	rawPlugins, lastkey, err := auth.ListPlugins(ctx, int(req.Limit), req.StartKey, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	encodedPlugins := make([]*types.PluginV1, 0, len(rawPlugins))
+	for _, rawPlugin := range rawPlugins {
+		plugin, ok := rawPlugin.(*types.PluginV1)
+		if !ok {
+			log.Warnf("Skipping unexpected plugin type %T, expected %T.", rawPlugin, plugin)
+			continue
```

Comment:
Continuing here felt odd at first, I was expecting an error.
However, after looking at other methods it looks like we are not consistent with how we treat item errors when building the list of resources.

Is there a reason for us to `continue` here instead of returning an error?
Just curious about this one, because we don't seem to have a clear pattern (unless I'm missing something obvious)